src/GafferScene/CameraQuery.cpp

```diff
@@ -0,0 +1,551 @@
+		{
+			if( const auto renderCameraData = scenePlug()->globals()->member<StringData>( g_renderCameraOption ) )
+			{
+				location = renderCameraData->readable();
+			}
+		}
+		else
+		{
+			location = locationPlug()->getValue();
+		}
+
+		if( !location.empty() )
+		{
+			const ScenePlug::ScenePath locationPath = ScenePlug::stringToPath( location );
+			ScenePlug::PathScope scope( context, &locationPath );
+			if( scenePlug()->existsPlug()->getValue() )
+			{
+				h.append( scenePlug()->objectHash( locationPath ) );
+			}
+		}
+	}
+	else if( outPlug()->isAncestorOf( output ) )
+	{
+		const ValuePlug *oPlug = outPlug( output );
+
+		if( output == oPlug->getChild( g_sourcePlugIndex ) )
+		{
+			internalParametersPlug()->hash( h );
+			const StringPlug *childQueryPlug = queryPlug( output );
+			childQueryPlug->hash( h );
+		}
+		else if(
+			oPlug->getChild( g_valuePlugIndex )->isAncestorOf( output ) ||
+			output == oPlug->getChild( g_valuePlugIndex )
+		)
+		{
+			internalParametersPlug()->hash( h );
+			const StringPlug *childQueryPlug = queryPlug( output );
+			childQueryPlug->hash( h );
+		}
+	}
+
+	ComputeNode::hash( output, context, h );
+}
+
+void CameraQuery::compute( Gaffer::ValuePlug *output, const Gaffer::Context *context ) const
+{
+	if( output == internalParametersPlug() )
+	{
+		IECore::CompoundDataPtr parameters = new IECore::CompoundData;
+
+		std::string location = "";
+		if( cameraModePlug()->getValue() == (int)CameraMode::RenderCamera )
+		{
+			if( const auto cameraData = scenePlug()->globals()->member<StringData>( g_renderCameraOption ) )
+			{
+				location = cameraData->readable();
+			}
+		}
+		else
+		{
+			location = locationPlug()->getValue();
+		}
+
+		if( !location.empty() )
+		{
+			const ScenePlug::ScenePath locationPath = ScenePlug::stringToPath( location );
+			ScenePlug::PathScope scope( context, &locationPath );
+			if( scenePlug()->existsPlug()->getValue() )
+			{
+				if( const auto camera = runTimeCast<const IECoreScene::Camera>( scenePlug()->objectPlug()->getValue() ) )
+				{
+					auto cameraWithGlobals = camera->copy();
+					SceneAlgo::applyCameraGlobals( cameraWithGlobals.get(), scenePlug()->globals().get(), scenePlug() );
+					for( const auto &p : cameraWithGlobals->parameters() )
+					{
+						if( p.first == g_fStop || p.first == g_shutter )
+						{
+							// We handle these specially below.
+							continue;
+						}
+
+						parameters->writable()[p.first] = new CompoundData( {
+							{ g_source, camera->parameters().count( p.first ) ? g_sourceCamera : g_sourceGlobals },
+							{ g_value, p.second }
+						} );
+					}
+
+					// SceneAlgo::applyCameraGlobals overrides fStop to 0.0 if `depthOfField` is False,
+					// so we always return the original camera fStop instead.
+					parameters->writable()[g_fStop] = new CompoundData( {
+						{ g_source, camera->parameters().count( g_fStop ) ? g_sourceCamera : g_sourceFallback },
+						{ g_value, new FloatData( camera->getFStop() ) }
+					} );
```

Comment:
It seems `applyCameraGlobals()` also doesn't call `camera->setDepthOfField()`, so a `depthOfField` query is broken if the source should come from the globals. I think maybe the best fix is just to make `applyCameraGlobals()` copy it over (and maybe even stop setting `fStop`)?

Reply:
Another way `applyCameraGlobals()` is being unhelpful is that it doesn't update the overscan values if overscan is off. So it's impossible to query something like `overscanLeft` unless you've already turned overscan on somewhere. Actually, it seems to be a bit weirder than that even, but I haven't quite figured out the specifics of what's happening.

I think the general problem here is that `applyCameraGlobals()` seems to be trying to resolve final values for rendering as if the renderer backends didn't respect the boolean toggles for overscan/depthOfField on/off. But we just want to query the values they have currently.

Reply:
> I think maybe the best fix is just to make applyCameraGlobals() copy it over (and maybe even stop setting fStop)?

Updated applyCameraGlobals to set a `depthOFfield` parameter on the camera in place of overriding `fStop` to 0.0 and updated the various renderer backends in 78b8b6a59f2955f2c6cd8c32141b038f85e0a8d8. Removed CameraQuery's special handling of `fStop` in bb2101adbc4813308f4a41d73b05e160f2573892.

> Another way applyCameraGlobals() is being unhelpful is that it doesn't update the overscan values if overscan is off.

Updated applyCameraGlobals to always apply `overscan[Top/Bottom/Left/Right]` to the camera when they exist in the globals in f1e0c170cf7cebafb34683f3ef19b2aa76a5befc.